after the release is before the release.

fkr opened the pull request.

fkr committed:
after the release is before the release.

Signed-off-by: Felix Kronlage-Dammers <fkr@hazardous.org>

garloff committed:
Merge branch 'main' into feat/r8

jschoone committed:
feat(R8): Add initial KaaS release notes

Signed-off-by: Jan Schoone <jan.schoone@uhurutec.com>

matofeder committed:
Add upgrade/migration notes about registry and monitoring

Signed-off-by: Matej Feder <matej.feder@dnation.cloud>

garloff commented:
Thanks for the contributions, @jschoone, @matofeder.
I think release notes should always have the reader in mind.

>     Cluster Stacks SCS to use Multi Stage Addons

So what? What is the benefit?
(I guess it is that with them, we can safely do rolling upgrades when newer k8s needs newer add-ons, as they are now sequenced correctly. But I'm guessing only, here ...)

>     ORC Image kind in Cluster Stacks SCS

Benefit: Using the new capi-upstream way to get the needed node images, reducing divergence and specific code in our Cluster Stacks on OpenStack implementation.

>     Helm charts to manage Cluster Stack Operator
>         simplifies configuration of release source
>         additional RBAC for provider specific resources can be configured

This one is good, I think.

Do we have cluster stacks that are working with latest v1.31 and v1.32 k8s?
That would be worth mentioning, no?

garloff commented:
> - The container registry has been upgraded to Harbor v2.12.2
>   - The backup and restore procedure was successfully tested with production data during the migration from SCS1 to SCS2 environment

Any features of v2.12 worth mentioning? Enhanced security scans? ...

> - The observability platform has been upgraded to dN Kubernetes Monitoring Stack v3.8.1
>   - It has been deployed in the new SCS2 environment, providing visibility into the SCS environment

Dito: Anything worth mentioning?

Also, we mention that we have migrated those systems. As a non-expert reader I would conclude that harbor and dN k8s monitoring stack are SCS internal tools. We should mention that these are tools that we expect to be useful to operators to run and we deploy them as well (drinking our own champaign).

matofeder committed:
Enhance upgrade/migration notes about registry and monitoring

Signed-off-by: Matej Feder <matej.feder@dnation.cloud>

matofeder commented:
> > * The container registry has been upgraded to Harbor v2.12.2
> >   
> >   * The backup and restore procedure was successfully tested with production data during the migration from SCS1 to SCS2 environment
> 
> Any features of v2.12 worth mentioning? Enhanced security scans? ...
> 
> > * The observability platform has been upgraded to dN Kubernetes Monitoring Stack v3.8.1
> >   
> >   * It has been deployed in the new SCS2 environment, providing visibility into the SCS environment
> 
> Dito: Anything worth mentioning?
> 
> Also, we mention that we have migrated those systems. As a non-expert reader I would conclude that harbor and dN k8s monitoring stack are SCS internal tools. We should mention that these are tools that we expect to be useful to operators to run and we deploy them as well (drinking our own champaign).

Thank you for the review. I agree, the enhanced version: 687ed0bfc8040cdb40001db7102ce724477a2227

matofeder committed:
Clarify Cluster Stacks  to use Multi Stage Addons

Signed-off-by: Matej Feder <matej.feder@dnation.cloud>

matofeder commented:
I tried to clarify the `Cluster Stacks SCS to use Multi Stage Addons` item 8120eb404b55f3bc44ffac5b4b746e9a6deebf49. @jschoone, please adjust if needed

berendt committed:
Add OSISM features and removals

Signed-off-by: Christian Berendt <berendt@osism.tech>

matofeder committed:
Add ORC installation as a requirement for the KaaS R7 → R8 upgrade

Signed-off-by: Matej Feder <matej.feder@dnation.cloud>

matofeder committed:
Add ORC Image kind in Cluster Stacks  explanation

Signed-off-by: Matej Feder <matej.feder@dnation.cloud>

matofeder committed:
Add SCS release version: v8.0.0 to the container registry and observability platform

Signed-off-by: Matej Feder <matej.feder@dnation.cloud>

garloff committed:
Improve SLURP language. Skipping .2 is possible.

The prior language could have been misread as upgrading to 2024.2 would
somehow prevent to use SLURP later (to skip 2025.2 for example).

Signed-off-by: Kurt Garloff <kurt@garloff.de>

garloff committed:
Move harbor and monitoring to operator section, add os-hm there.

Signed-off-by: Kurt Garloff <kurt@garloff.de>

garloff committed:
Mention Domain-Manager that we contributed.

Also: OpenStack names, mention k8s-1.32.

Signed-off-by: Kurt Garloff <kurt@garloff.de>

jschoone committed:
docs(R8): Mention additional RBAC for CSO to manage ORC CRDs

Signed-off-by: Jan Schoone <jan.schoone@uhurutec.com>

jschoone committed:
chore(r8): Add Cluster Stack releases

Signed-off-by: Jan Schoone <jan.schoone@uhurutec.com>

garloff committed:
Add Scope, Tags, Issues, Compliance, Future, Thanks.

With this, the RN8 look complete to me.

Signed-off-by: Kurt Garloff <kurt@garloff.de>

garloff commented:
Two things for discussion:
1. I mention two testcases that fail for network policies with k8s with cilium CNI. Please review for correctness. (Was it exactly two cases? Can we maybe even mention or reference them?) @matofeder, @jschoone ?
2. I mention the intention of OSISM to publish a release with OpenStack 2025.1 earlier than Sep/Oct. @berendt, please confirm that we did not misunderstand you and that the statement is correct. (Note that having the intention but not announcing it is not a good option -- some predictability is a key value we create.)

garloff commented:
Oh, we should merge this before 9am CEST tomorrow, so it's there when the PR goes out.
Approvals welcome -- even if we do a minor correction after the initial merge.
(I am aware that I can technically do the approval myself, but I'd prefer not to do this when having done the last significant additions.)

fkr commented on Release8.md:
```diff
@@ -0,0 +1,185 @@
+
+## Scope
+
+SCS continues to deliver a coordinated release cvoering the various modules that comprise
```
```suggestion
SCS continues to deliver a coordinated release covering the various modules that comprise
```

fkr commented on Release8.md:
```diff
@@ -0,0 +1,185 @@
+SCS continues to deliver a coordinated release cvoering the various modules that comprise
+the SCS reference implementation. SCS includes the latest proven upstream technologies and
+ensures that upgrades to R8 are smooth using the CI processes. The reference implementation
+with the default settings fulfills the requirements from SCS-compatibel IaaS v5.1 and
```
```suggestion
with the default settings fulfills the requirements from SCS-compatible IaaS v5.1 and
```

fkr commented on Release8.md:
```diff
@@ -0,0 +1,185 @@
+
+- Cluster API Provider OpenStack v0.12
+  - adds OpenStack Resource Controller (ORC) as external component
+- OpenStack 2024.2 is included in R8, but can be skipped in case Operators
+  want to make use of skipping over it and do directly from 2024.1 to 2025.1
```
```suggestion
- OpenStack 2024.2 is included in R8, but can be skipped in case Operators
  want to make use of skipping over it and upgrade directly from 2024.1 to 2025.1
```

fkr commented on Release8.md:
```diff
@@ -0,0 +1,185 @@
+  Netbox Manager that allows you to manage content from the Netbox with an independent repository
+- There is now a special burn-in image for preparing new hardware (this is used when new cloudpods
+  are created and Ironic cannot yet be used)
+- A bunch of new services like such as Wazuh agent, Teleport agent or Dnsmasq
```
```suggestion
- A bunch of new services such as Wazuh agent, Teleport agent or Dnsmasq
```

fkr commented on Release8.md:
```diff
@@ -0,0 +1,185 @@
+
+We appreciate contribution to strategy and implementation, please join
+our community -- or just leave input on the github issues and PRs.
+Have a look at our [How to contribute page](https://scs.community/contribute/).
```
```suggestion
Have a look at our [How to contribute page](https://docs.scs.community/community/).
```

fkr commented on Release8.md:
```diff
@@ -0,0 +1,185 @@
+elected members of the project board have stepped up and taken responsibility to drive
+the efforts to develop the software. We'd like to call out dNation, OSISM, UhuruTec,
+syself, syseleven and S7n Cloud Services for their contributions. We'd also like to
+thank ScaleUp Technologies, PlusServer, Wavecon, Artcodix, and Cleura for sponsoring
```
```suggestion
thank ScaleUp Technologies, plusserver, Wavecon, Artcodix, and Cleura for sponsoring
```

fkr reviewed (commented):
please see comments

garloff committed:
Typo

Co-authored-by: Felix Kronlage-Dammers <fkr@hazardous.org>
Signed-off-by: Kurt Garloff <kurt@garloff.de>

garloff committed:
Update Release8.md

Co-authored-by: Felix Kronlage-Dammers <fkr@hazardous.org>
Signed-off-by: Kurt Garloff <kurt@garloff.de>

garloff committed:
Update Release8.md

Co-authored-by: Felix Kronlage-Dammers <fkr@hazardous.org>
Signed-off-by: Kurt Garloff <kurt@garloff.de>

garloff committed:
Update Release8.md

Co-authored-by: Felix Kronlage-Dammers <fkr@hazardous.org>
Signed-off-by: Kurt Garloff <kurt@garloff.de>

garloff committed:
Update Release8.md

Co-authored-by: Felix Kronlage-Dammers <fkr@hazardous.org>
Signed-off-by: Kurt Garloff <kurt@garloff.de>

garloff committed:
Update Release8.md

Co-authored-by: Felix Kronlage-Dammers <fkr@hazardous.org>
Signed-off-by: Kurt Garloff <kurt@garloff.de>

garloff committed:
Add entry for R8 into the main README.

Signed-off-by: Kurt Garloff <kurt@garloff.de>

garloff committed:
Merge branch 'feat/r8' of github.com:SovereignCloudStack/release-notes into feat/r8
